src/App.tsx

```diff
@@ -1,14 +1,72 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, {
+  useEffect,
+  useState,
+  useContext,
+  useMemo,
+} from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { TodoContext } from './components/TodoContext';
+import { Filter, TodoStates } from './types/Filter';
+
+const DEFAULT_FILTER: Filter = {
+  select: TodoStates.All,
+  input: '',
+};
+
+const getFiltredTodos = (filter: Filter, todos: Todo[]): Todo[] => {
+  const { input, select } = filter;
+
+  if (!input && select === TodoStates.All) {
+    return todos;
+  }
+
+  const validatedInput = input.toLowerCase().trim();
+  let newTodos = todos
+    .filter(({ title }) => title.toLowerCase().includes(validatedInput));
+
+  switch (select) {
+    case TodoStates.Active:
+      newTodos = newTodos.filter(({ completed }) => !completed);
+      break;
+    case TodoStates.Completed:
+      newTodos = newTodos.filter(({ completed }) => completed);
+      break;
+    default:
+  }
```

Comment:
Nothing happens in the `default` case.
Also, You can filter just one time instead of filtering by query and completed separately